src/riscv/lib/src/machine_state.rs

```diff
@@ -328,10 +330,24 @@ impl<MC: memory::MemoryConfig, BCC: BlockCacheConfig, B: Block<MC, M>, M: backen
     where
         M: backend::ManagerReadWrite,
     {
-        self.core.reset();
+        self.core.reset(NoopMemoryGovernanceListener);
         self.block_cache.reset();
     }
 
+    /// Get access to the main memory, with a listener to hook into any permission updates.
+    ///
+    /// This is required to keep parts of [`MachineState`], that do not form part of the PVM state,
+    /// synchronised with main memory so as to ensure determinism regardless of whether these
+    /// additional parts of the state are populated or not.
+    pub(crate) fn memory_with_listener(
+        &mut self,
+    ) -> (
+        &mut MC::State<M>,
+        impl MemoryGovernanceListener + use<MC, BCC, B, M>,
```

Comment:
The code compiled just fine by eliminating the `+ use<MC, BCC, B, M>`. I knew that this is usually used for lifetimes. What would be the purpose here? (I'm fairly new to the `use` clause)

Reply:
removed the `use` constraint [here](https://github.com/tezos/riscv-pvm/compare/7ea21465fedbb4c74e1b35f38380812a097345ee..4f50efeffe7519f314c4cf9f68d70678ca8bd09a)